contracts/BurningManager.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity ^0.6.2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./Promos.sol";
+import "./util.sol";
+import "./Garrison.sol";
+import "./cryptoblades.sol";
+
+contract BurningManager is Initializable, AccessControlUpgradeable {
+    using SafeMath for uint256;
+
+    bytes32 public constant GAME_ADMIN = keccak256("GAME_ADMIN");
+    bytes32 public constant BURNER_ROLE = keccak256("BURNER_ROLE");
+
+    // STATE
+    Characters public characters;
+    CryptoBlades public game;
+    Garrison public garrison;
+
+    function initialize(Characters _characters, Garrison _garrison, CryptoBlades _game)
+        public
+        initializer
+    {
+        __AccessControl_init();
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+
+        characters = _characters;
+        garrison = _garrison;
+        game = _game;
+    }
+
+    // MODIFIERS
+
+    modifier isCharacterOwner(uint256 burnId) {
+        _isCharacterOwner(burnId);
+        _;
+    }
+
+    function _isCharacterOwner(uint256 burnId) internal view {
+        require(hasRole(BURNER_ROLE, msg.sender) || characters.ownerOf(burnId) == msg.sender || garrison.characterOwner(burnId) == msg.sender, 'Not owner');
```

Comment:
this role check might not make sense. This contract is the source of calls on the soul burns in characters (which are restricted), and the only thing that should have this role (afaik) is the market

Reply:
Only market has this role. What's a better way of checking if it's the market making the call?

Reply:
right, i forgot this was a logical AND statement. So not technically an issue. But it's certainly a loaded check when we're iterating over multiple characters.
I guess for now it's fine to leave it as is, unless a smart solution presents itself. Alternatively you can have an extra burn function that is just for the role.